proposals/gc/MVP.md

```diff
@@ -737,6 +759,9 @@ The opcode for heap types is encoded as an `s33`.
 | 0xfb15 | `array.get_u $t` | `$t : typeidx` |
 | 0xfb16 | `array.set $t` | `$t : typeidx` |
 | 0xfb17 | `array.len` | `_ : u32` (TODO: remove, was typeidx) |
+| 0xfb19 | `array.new_fixed $t N` | `$t : typeidx`, `N : u32` |
+| 0xfb1d | `array.new_data $t $d` | `$t : typeidx`, `$d : dataidx` |
+| 0xfb1e | `array.new_elem $t $e` | `$t : typeidx`, `$e : elemidx` |
```

Comment:
@manoskouk, does the current `array.init_from_data` instruction in V8 work for both data and element segments or just for data segments?

Having two separate instructions for this seems reasonable and I like the new names, but could we choose a different opcode for `array.new_elem`? The problem with the current one is that it overlaps with V8's `array.init_from_data_static`, so implementing the new instruction with the opcode here will require a binary format back compat break in V8.

Reply:
The current `array.new_from_data` in v8 supports only numeric arrays and data segments. Regarding `array.new_from_elem`, the discussion on #260 did not come up with a decision regarding how to handle cyclic dependencies between the element section and constant uses of `array.new_from_elem`, so we postponed its implementation. We could, of course, introduce it as a non-constant-only expression for now.

Reply:
@tlively, ah, sorry, I had missed the difference in operands. Changed opcodes to 0x1b and 0x1c, which seem to be free right now.

@manoskouk, thanks for the reminder. So far, the proposal has not declared any of the allocation instructions constant, although we discussed it. I suppose there is agreement to change this, so I went ahead and incorporated that into this PR. And I agree that, for the time being, excluding `array.new_elem` from being constant is the simplest interim solution. Once we have introduced repeated sections, we can relax that.

Reply:
FWIW, having these instructions as constant is essential for the dart2wasm use case of using them for constants. Including `array.new_elem` is necessary to not have a limit on how big `List` constants can be.

Reply:
`array.new_fixed` works fine as long as the created array does not exceed the maximum number of operands for `array.new_fixed`. For longer arrays, `array.new_elem` is currently the only way to create them as constant expressions.